restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -502,46 +527,59 @@ private void invite(final Object message) throws IOException, NumberParseExcepti
+                                    sendNotification(errMsg, 11001, "warning", true);
+                                    final SipServletResponse resp = request.createResponse(SC_NOT_FOUND, "Cannot complete P2P call");
+                                    resp.send();
+                                }
+                            } else {
+                                //Extensions didn't allowed this call
+                                if (logger.isDebugEnabled()) {
+                                    final String errMsg = "Client not Allowed to make this outbound call";
+                                    logger.debug(errMsg);
+                                }
+                                String errMsg = "Cannot Connect to Client: " + toClient.getFriendlyName()
+                                        + " : Make sure the Client exist or is registered with Restcomm";
+                                sendNotification(errMsg, 11001, "warning", true);
+                                final SipServletResponse resp = request.createResponse(SC_FORBIDDEN, "Call not allowed");
+                                resp.send();
+                            }
+                            ec.executePostOutboundAction(er, extensions);
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
                         }
-                        // if all goes well with proxying the invitation on to the next client
-                        // then we can end further processing of this INVITE
-                    } else {
-                        String errMsg = "Cannot Connect to Client: " + toClient.getFriendlyName()
-                                + " : Make sure the Client exist or is registered with Restcomm";
-                        sendNotification(errMsg, 11001, "warning", true);
-                        final SipServletResponse resp = request.createResponse(SC_NOT_FOUND, "Cannot complete P2P call");
-                        resp.send();
-                    }
-                } else {
-                    //Extensions didn't allowed this call
-                    if (logger.isDebugEnabled()) {
-                        final String errMsg = "Client not Allowed to make this outbound call";
-                        logger.debug(errMsg);
                     }
-                    String errMsg = "Cannot Connect to Client: " + toClient.getFriendlyName()
-                            + " : Make sure the Client exist or is registered with Restcomm";
-                    sendNotification(errMsg, 11001, "warning", true);
-                    final SipServletResponse resp = request.createResponse(SC_FORBIDDEN, "Call not allowed");
-                    resp.send();
-                }
-                ec.executePostOutboundAction(er, this.extensions);
+                }, system.dispatcher());
```

Comment:
@agafox since the Push Notification is a task not related to Akka actors etc, maybe we should create new Execution Context or use the one created for the S3 upload, and use it here to execute this task.